### PR TITLE
Add support for building Swift apps

### PR DESCRIPTION
**Pull request by lvillani**
The .ipa of a Swift application must contain a "SwiftSupport" directory
with all the runtime libraries required by Swift (even though they are
already included in the application bundle's "Frameworks" directory).

This step is NOT done by Xcode. The "Organizer" does that when someone
exports the application for distribution to iTunes Connect.

By inspecting the .ipa produced by the Organizer, we see that the SHA1
of all Swift runtime libraries in the "SwiftSupport" directory
are different from the ones inside the app bundle's "Frameworks"
directory. They seem to come from Xcode's toolchain directory instead.

This commit tries to match the behavior of the Organizer by doing the
following (after the PackageApplication step):

- Inspect the "Frameworks" directory within the app bundle to determine
  which runtime libraries we need (if any)
- Copy all Swift runtime libraries from Xcode's toolchain directory in
  a temporary directory with the required directory structure.
- Append the temporary "SwiftSupport" directory to the .ipa file

### CODE REVIEW

File: lib/shenzhen/commands/build.rb [hunk 1]
```diff
@@ -87,6 +87,36 @@
...
+        xcode = `xcode-select --print-path`.strip
+
+        @ipa_swift_frameworks.each do |path|
+          framework = File.basename(path)
+
+          FileUtils.copy_file("#{xcode}/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/#{@sdk}/#{framework}", File.join(swift_support, framework))
+        end
+
+        # Add "SwiftSupport" to the .ipa archive
+        pwd = Dir.pwd
```
Comment (KrauseFx) [inline lib/shenzhen/commands/build.rb]: You could change it to make use of Ruby blocks:
```ruby
Dir.chdir(tmpdir) do
   abort unless system %{zip --recurse-paths "#{@ipa_path}" "SwiftSupport" #{'> /dev/null' unless $verbose}}
end
```

Comment (lvillani) [inline lib/shenzhen/commands/build.rb]: Right! I didn't see that while looking up the docs. I know enough Ruby to be dangerous but it's not really my cup of tea :)

**Comment by lvillani** [comment]
I updated the pull request to include improvements reported by @KrauseFx

**Comment by mattt** [comment]
Amazing work, @lvillani. Thanks so much for your PR! Having Swift support is a huge deal to a lot of folks, myself included.

I'm cutting a new minor release now.

**Comment by KrauseFx** [comment]
Thanks for merging @mattt, this is big! :+1:

**Pull request merged**